src/WebJobs.Extensions.DurableTask/Listener/OutOfProcOrchestrationShim.cs

```diff
@@ -104,6 +125,195 @@ internal async Task<bool> ScheduleDurableTaskEvents(OrchestrationInvocationResul
             return true;
         }
 
+        /// <summary>
+        /// Invokes a DF API based on the input action object.
+        /// Most APIs will return a Task to be awaited, but others such as SignalEntity do not.
+        /// </summary>
+        /// <param name="action">An OOProc action object representing a DF task.</param>
+        /// <returns>If the API returns a task, the DF task corresponding to the input action. Else, null.</returns>
+        private Task InvokeAPIFromAction(AsyncAction action)
+        {
+            Task task = null;
+            switch (action.ActionType)
```

Comment:
It is almost certainly identical. I did not want to modify the pre-existing `ProcessAsyncActions` to avoid accidentally breaking backwards compatibility, but you're right in that there's a wasted opportunity for code re-use in here.

I'll look into that refactor.